akka-typed/src/main/java/akka/typed/javadsl/Actor.java

```diff
@@ -164,23 +164,21 @@ else if (Behavior.isAlive(behv))
    * {@link ActorContext} that allows access to the system, spawning and watching
    * other actors, etc.
    * 
-   * In either case—signal or message—the next behavior must be returned. If no
-   * change is desired, use {@link #same}.
+   * This constructor is called stateful because processing the next message
+   * results in a new behavior that can potentially be different from this one.
+   * If no change is desired, use {@link #same}.
    * 
-   * @param signal
-   *          the function that describes how this actor reacts to the given
-   *          signal
    * @param message
    *          the function that describes how this actor reacts to the next
    *          message
    * @return the behavior
    */
-  static public <T> Behavior<T> signalOrMessage(Function2<ActorContext<T>, Signal, Behavior<T>> signal,
-      Function2<ActorContext<T>, T, Behavior<T>> message) {
-    return new Stateful<T>(signal, message);
+  static public <T> Behavior<T> stateful(Function2<ActorContext<T>, T, Behavior<T>> message) {
```

Comment:
This signature contradicts the first sentence of the JavaDoc (i.e. the first sentence must be swapped with that for the other signature).